google/cloud/aiplatform/jobs.py

```diff
@@ -15,10 +15,214 @@
+            batch_prediction_job_name (str):
+                Required. A fully-qualified BatchPredictionJob resource name or ID.
+                Example: "projects/.../locations/.../batchPredictionJobs/456" or
+                "456" when project and location are initialized or passed.
+            project: Optional[str] = None,
+                Optional project to retrieve BatchPredictionJob from. If not set,
+                project set in aiplatform.init will be used.
+            location: Optional[str] = None,
+                Optional location to retrieve BatchPredictionJob from. If not set,
+                location set in aiplatform.init will be used.
+            credentials: Optional[auth_credentials.Credentials] = None,
+                Custom credentials to use. If not set, credentials set in
+                aiplatform.init will be used.
+        """
+        valid_batch_prediction_job_name = full_resource_name(
+            resource_name=batch_prediction_job_name,
+            resource_noun="batchPredictionJobs",
+            project=project,
+            location=location,
+        )
+
+        super().__init__(
+            valid_job_name=valid_batch_prediction_job_name,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    def iter_outputs(
+        self, bq_query_limit: Optional[int] = 100
+    ) -> Iterable[Union[storage.Blob, bigquery.job.QueryJob]]:
+        """Returns an Iterable object to traverse the output files, either a list
+        of GCS Blobs or a BigQuery QueryJob depending on the output config set
+        when the BatchPredictionJob was created.
+
+        Args:
+            bq_query_limit: Optional[int] = 100
+                Limit on rows to select from prediction table in BigQuery dataset.
+                Only used when retrieving predictions from a bigquery_destination_prefix.
+                Default is 100.
+
+        Returns:
+            Iterable[Union[storage.Blob, bigquery.job.QueryJob]]:
+                Either a list of GCS Blob objects within the prediction output
+                directory or an iterable QueryJob with predictions.
+
+        Raises:
+            RuntimeError:
+                If BatchPredictionJob is in a JobState other than SUCCEEDED,
+                since outputs cannot be retrieved until the Job has finished.
+            NotImplementedError:
+                If BatchPredictionJob succeeded and output_info does not have a
+                GCS or BQ output provided.
+        """
+
+        job_status = self.status()
+
+        if job_status != JobState.JOB_STATE_SUCCEEDED:
+            raise RuntimeError(
+                f"Cannot read outputs until BatchPredictionJob has succeeded, "
+                f"current status: {job_status}"
+            )
+
+        output_info = self._gca_resource.output_info
+
+        # GCS Destination, return Blobs
+        if output_info.gcs_output_directory not in ("", None):
```

Comment:
maybe just `if not output_info.gcs_output_directory`?

Reply:
Unfortunately the default for `output_info` has its fields set to empty string, both GCS and BQ. Another option is to look at `output_config.predictions_format` to see if it's `"bigquery"`, `"jsonl"`, `"csv"`, etc.